crates/iota-sdk-types/src/iota_names/error.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::ObjectId;
+
+#[derive(thiserror::Error, Debug, Clone, Eq, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde_derive::Serialize, serde_derive::Deserialize)
+)]
+pub enum IotaNamesError {
+    #[error("Name length {0} exceeds maximum length {1}")]
+    NameLengthExceeded(usize, usize),
+    #[error("Label length {0} outside of valid range [{1}, {2}]")]
+    InvalidLabelLength(usize, usize, usize),
+    #[error("Hyphens are not allowed as first or last character of a label")]
+    HyphensAsFirstOrLastLabelChar,
+    #[error(
+        "Only lowercase letters, numbers, and hyphens are allowed as label characters. Got \'{0}\' at position {1}"
+    )]
+    InvalidLabelChar(char, usize),
+    #[error("Name must contain at least two labels, TLN and SLN")]
+    NotEnoughLabels,
+    #[error("Name must include only one separator")]
+    InvalidSeparator,
```

Comment:
tbh this error is a bit confusing since we actually have two separators and it's not clear that this error and its message make only sense for the `@` separator. 

Can we be more explicit here?